core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -458,6 +462,15 @@ object SparkSubmit {
       }
     }
 
+    if (isMesosCluster) {
+      // Mesos cluster dispatcher only supports the REST interface
```

Comment:
maybe we should `assert(args.useRest, "Mesos cluster mode is only supported through the REST submission API")`